staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -489,6 +493,8 @@ func (cnc *CloudNodeController) syncNode(ctx context.Context, nodeName string) e
 			return err
 		}
 
+		removeCloudProviderTaintDelay.Observe(time.Since(newNode.ObjectMeta.CreationTimestamp.Time).Seconds())
```

Comment:
I like.

One small issue is we get nothing if there is a failure with the update or any of the precursor steps. Would be good to have a nodeInitializeFailure metric.